### PR TITLE
DOP-4919: add how-to structured data to procedures

### PR DESCRIPTION
**Pull request by seungpark**
### Stories/Links:

DOP-4919

### Current Behavior:

Current procedures contain now structured data

### Staging Links:

[Page with nested procedures](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/cloud-docs/seung.park/DOP-4919/data-federation/query/sql/getting-started/get-started-advanced/index.html)
[Schema markup validator of above](https://validator.schema.org/#url=https%3A%2F%2Fdocs-mongodb-org-stg.s3.us-east-2.amazonaws.com%2Fmaster%2Fcloud-docs%2Fseung.park%2FDOP-4919%2Fdata-federation%2Fquery%2Fsql%2Fgetting-started%2Fget-started-advanced%2Findex.html)

[Page with procedures](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/master/cloud-docs/seung.park/DOP-4919/atlas-search/index.html)
[Schema validator ](https://validator.schema.org/#url=https%3A%2F%2Fdocs-mongodb-org-stg.s3.us-east-2.amazonaws.com%2Fmaster%2Fcloud-docs%2Fseung.park%2FDOP-4919%2Fatlas-search%2Findex.html)

### Notes:
- Google has marked the How To structured data as deprecated, so we are de-prioritizing `name` and `image` properties for How To structured data. This also disables from [rich results tester](https://search.google.com/test/rich-results/result?id=RRkk60hBFj0aLxlL36S9tw) being able to collect the Structured data from these staging links. Included the Schema.org validator instead above

### README updates

- - [ ] This PR introduces changes that should be reflected in the README, and I have made those updates.
- - [X] This PR does not introduce changes that should be reflected in the README

### CODE REVIEW

**Comment by seungpark** [comment]
Dealing with a bug where at build time some characters are coming in as escaped characters. Suspecting this is due to use of quotes within step's texts but will verify tomorrow 👍

**Comment by seungpark** [comment]
> Dealing with a bug where at build time some characters are coming in as escaped characters. Suspecting this is due to use of quotes within step's texts but will verify tomorrow 👍

using a script tag with dangerouslySetInnerHtml property [addressed this](https://github.com/mongodb/snooty/pull/1255/commits/681ef17d95683748d1ce2be4f1c290e4bfa7aa7d#diff-308534dbe97ea5b8a30cd1cbb7bc49e089331ae0495801e8ec2f41dc98ef2cb1R77-R79). thanks @rayangler for the sanity check!

**Review by rayangler — COMMENTED** [review]
Looks great!

**Review by mmeigs — COMMENTED** [review]
Thanks for the comments! Will help a lot in the future.
Also, is it expected that the validation[ is failing](https://validator.schema.org/#url=https%3A%2F%2Fdocs-mongodb-org-stg.s3.us-east-2.amazonaws.com%2Fmaster%2Fcloud-docs%2Fseung.park%2FDOP-4919%2Fatlas-search%2Findex.html)?

File: src/components/Procedure/index.js [hunk 1]
```diff
@@ -47,18 +52,38 @@ const getSteps = (children) => {
...
-    <StyledProcedure procedureStyle={style}>
-      {steps.map((child, i) => (
-        <Step {...rest} nodeData={child} stepNumber={i + 1} stepStyle={style} key={i} />
-      ))}
-    </StyledProcedure>
+    <AncestorComponentContextProvider component={'procedure'}>
+      {howToSd && (
+        // using dangerouslySetInnerHTML as JSON is rendered with
+        // encoded quotes at build time
+        <script id={`how-to-sd-${id}`} type="application/ld+json" dangerouslySetInnerHTML={{ __html: howToSd }} />
```
[the remark this answers is not in the document]
Comment (seungpark) [reply src/components/Procedure/index.js:79]: good point! don't believe we need `id` attribute for anything. removed

**Comment by seungpark** [comment]
> Thanks for the comments! Will help a lot in the future. Also, is it expected that the validation[ is failing](https://validator.schema.org/#url=https%3A%2F%2Fdocs-mongodb-org-stg.s3.us-east-2.amazonaws.com%2Fmaster%2Fcloud-docs%2Fseung.park%2FDOP-4919%2Fatlas-search%2Findex.html)?

I believe some of the HTML markup is missing but the HowTo can still be validated on the right!
image

**Review by mmeigs — APPROVED** [review]
LGTM

**Review by rayangler — APPROVED** [review]

**Pull request merged**